src/net/parsers/texttracks/webvtt.js

```diff
@@ -0,0 +1,311 @@
+
+    if(spanElement.childNodes.length != 0) {
+      spanElement.appendChild(document.createElement("br"));}
+    
+    formatWebVTTtoHTML(linified[index], styleElements)
+      .forEach(child =>{
+        spanElement.appendChild(child);
+      });
+
+    index++;
+  }
+
+  region.appendChild(pElement) ;
+  pElement.appendChild(spanElement);
+
+  return { start: range.start, end: range.end, content: region };
+}
+
+function parseTimeCode(text) {
+  const both = text.split(" --> ");
+
+  const s = both[0]
+        .split(":").map(val => parseFloat(val));
+  const e = both[1].split(" ")[0].split(":")
+        .map(val => parseFloat(val));
+
+  const start = s[2] + s[1] * 60 + s[0] * 60 * 60;
+  const end = e[2] + e[1] * 60 + e[0] * 60 * 60;
+
+  return { start, end };
+}
+ 
+  /**
+  * Format WebVTT tags and classes into usual HTML.
+  * <b *> => <b>
+  * <u *> => <u>
+  * <i *> => <i>
+  * <c.class *> => <c.class>
+  * Style is inserted if associated to tag or class.
+  * @param {string} text 
+  * @param {Array<Object>} styleElements
+  */
+function formatWebVTTtoHTML(text, styleElements){
+
+  const classes = ["U","I","B"];
+  const styleClasses = styleElements.map(f => f.className.toUpperCase());
+  const filtered = text
+    // Remove timestamp tags
+    .replace(/<[0-9]{2}:[0-9]{2}.[0-9]{3}>/, "")
+    // Remove tag content or attributes (e.g. <b dfgfdg> => <b>)
+    .replace(/<([u,i,b,c])((?:\.){0,1}.*?)( .*?){0,1}>(.*?)<\/\1>/g, 
+      "<$1$2>$4</$1$2>")
+    // Remove all unauthorised tags (u,i,b,c.)
+    .replace(/<([^u,i,b,c])((?:\.){0,1}.*?)( .*?){0,1}>(.*?)<\/\1>/g, "$4");
+
+  const parser = new DOMParser();
+  const parsedWebVTT = parser.parseFromString(filtered, "text/html");
+  const nodes = parsedWebVTT.body.childNodes;
+
+    /**
+   * Apply styles to specifig tag in children nodes. 
+   * (e.g. If class "b" has style, then : <b style="content">
+   * )
+   * Change class tags into span with associated style, or text*
+   * First it was: <c.class>...</c>. Then <class></class>. 
+   * Finally <span style="content"></span> or text.
+   */
+  function parseNode(nodeToParse) {
+    const parsedNodeArray = [];
+    let i = 0;
+    for(i = 0; i < nodeToParse.length; i++){
+      parsedNodeArray[i] = setStyleForNode(nodeToParse);
+
+      // Parse again node if there is there are inner elements,
+      // other than text elements.
+      if(parsedNodeArray[i] && parsedNodeArray[i].childNodes.length > 1){
+        const innerNodeArray = parseNode(parsedNodeArray[i].childNodes);
+        while (parsedNodeArray[i].firstChild) {
+          parsedNodeArray[i].removeChild(parsedNodeArray[i].firstChild);
+        }
+        innerNodeArray.forEach(g => {
+          parsedNodeArray[i].appendChild(g);
+        });
+      }
+    }
+
+  /**
+   * Reduce node array on contiguous text tags.
+   * (e.g. <text><text><span><text> => <text><span><text>)
+   */
+    const reducedNodeArray = parsedNodeArray.reduce((a, b) => {
```

Comment:
Those variable names are difficult to follow.

Reply:
Also, I'm not sure of why you do 2 loops instead of one

Reply:
We don't need anymore this method.